### PR TITLE
New UI- Group Discription Page

### PR DESCRIPTION
**Pull request by darkshredder**
Fixes #

### Screenshots of the changes (If any) -

ezgif com-video-to-gif

ezgif com-video-to-gif-2

### CODE REVIEW

**Comment by darkshredder** [comment]
Any Suggestions??

**Comment by aayush-05** [comment]
-Fix hound issues
-Keep the styles in a separate file
-Are the designs responsive, I can't see the use of Bootstrap classes (Send a video while you shrink the browser window)

**Comment by darkshredder** [comment]
> -Fix hound issues
> -Keep the styles in a separate file
> -Are the designs responsive, I can't see the use of Bootstrap classes (Send a video while you shrink the browser window)

ezgif com-video-to-gif (1)
will do rest put in separate files

**Comment by aayush-05** [comment]
Please add atleast 4 members in the group, and then send the video.

**Comment by darkshredder** [comment]
> Please add atleast 4 members in the group, and then send the video.

ezgif com-video-to-gif (2)

Does this seem good?

**Comment by aayush-05** [comment]
Squeeze the width of the window to about 400px and then scroll up and down , so that I can have a look at the mobile view

**Comment by darkshredder** [comment]
> Squeeze the width of the window to about 400px and then scroll up and down , so that I can have a look at the mobile view

ezgif com-video-to-gif (3)

**Comment by aayush-05** [comment]
I don't think the content looks good on small screens, can you use bootstrap cards and align things accordingly to the screen size. (Like on smaller screens, Name could be on one line, deadline on another and so on).
image
image
Whatsay @nitin10s ?

**Comment by darkshredder** [comment]
sure will do it in some time

**Comment by nitin10s** [comment]
1. Put the remove button on the next row in mobile view.
2. We can't do much for the long text in Deadline column. Though you can add some margins between the Show, Edit, Destroy button.
3. Put the Add Member and Add new Assignment button on the next row in mobile view as well. 

Everything else looks good to me. Great Work.

**Comment by darkshredder** [comment]
> 1. Put the remove button on the next row in mobile view.
> 2. We can't do much for the long text in Deadline column. Though you can add some margins between the Show, Edit, Destroy button.
> 3. Put the Add Member and Add new Assignment button on the next row in mobile view as well.
> 
> Everything else looks good to me. Great Work.

ezgif com-video-to-gif (5)
ezgif com-video-to-gif (6)


This seems good enough to me should I commit it?

**Comment by nitin10s** [comment]
This is all good except the "Add new Assignment" button. It is actually out of place. Maybe increasing the Y-margins might help

**Comment by darkshredder** [comment]
Screenshot 2020-02-29 at 1 14 19 AM
Is it Okay??

**Comment by nitin10s** [comment]
Yes looks good but code climate has a lot of issues!

**Comment by darkshredder** [comment]
@aayush-05  @nitin10s  work is done I believe

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -9,8 +9,11 @@
 @import 'about';
 @import 'editor';
 @import 'groups';
+@import 'group_members';
 @import 'trumbowyg/dist/ui/trumbowyg.min.css';
 
+
+
```
Comment (tachyons) [inline app/assets/stylesheets/application.scss]: ✂️

File: app/views/groups/show.html.erb [hunk 2]
```diff
@@ -1,91 +1,69 @@
 <link rel="stylesheet" type="text/css" href="/css/tooltip.css">
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
```
Comment (tachyons) [inline app/views/groups/show.html.erb]: Use npm for assets, not cdn

File: app/views/groups/show.html.erb [hunk 3]
```diff
@@ -95,56 +73,52 @@
...
             <th>Grade</th>
           <% end %>
         </tr>
         </thead>
-
         <tbody>
         <tbody>
         <% @group.assignments.includes(projects: :grade).each do |assignment| %>
-            <tr>
+            <tr class="arows" style="background-color: #f8fffc;border-color: solid #bcdbd3;" >
```
Comment (tachyons) [inline app/views/groups/show.html.erb]: No inline styling please

File: app/views/groups/show.html.erb [hunk 4]
```diff
@@ -1,91 +1,69 @@
...
-          </table>
-
+      <br>
+      <br>
+      <h3>Members </h3>
+      <button type="button" style="color:white; background-color:#42b983 !important;" class="float-sm-left float-md-right float-lg-right float-xl-right addmem" data-toggle="modal" data-target="#myModal">Add member</button>
+      <br><br><br> 
+      <div class="row">
+        <% @group.users.each do |user| %>
+          <div class="card text-white mb-3" style="max-width: 18rem; background-color:#f8fffc; border: 1px solid #bcdbd3; margin:2px; ">
```
Comment (tachyons) [inline app/views/groups/show.html.erb]: Extract styles to css

**Comment by darkshredder** [comment]
> Remove inline styles and move to css, also improve the naming of css classes http://bdavidxyz.com/blog/how-to-name-css-classes/
@tachyons 
Can I do It later my exams has started

**Comment by nitin10s** [comment]
@darkshredder After 7 march would be too late. Can you please fix this in a couple of days.

**Comment by darkshredder** [comment]
I will try my best to complete it today night then as on Monday I have exam.

**Comment by darkshredder** [comment]
@nitin10s  @tachyons  I have updated it but there is only 1 inline styling as there is need to put important ,If I remove that the button disappears on hover as someone has applied some css on button

**Comment by aayush-05** [comment]
That bug has been fixed now, try removing it and see if it works now.

**Comment by darkshredder** [comment]
Yeah Now It works ..... Finally Done!!!

**Comment by nitin10s** [comment]
@tachyons, this looks good to me.

File: app/assets/stylesheets/group_members.scss [hunk 5]
```diff
@@ -1,3 +1,206 @@
...
+  border: 0;
+  border-radius: 5px;
+  color: $white;
+  cursor: pointer;
+  font-size: 16px;
+  left: 4px;
+  padding: 8px;
+  position: relative;
+  top: -7px;
+
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: ✂️

File: app/assets/stylesheets/group_members.scss [hunk 6]
```diff
@@ -1,3 +1,206 @@
...
+}
+
+.card-mem {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: It is not clear what is card-mem here, Can we have better name ?

File: app/assets/stylesheets/group_members.scss [hunk 7]
```diff
@@ -1,3 +1,206 @@
...
+  background-color: $gray-100;
+  border: 1px solid $green-border;
+  margin: 2px;
+  max-width: 18rem;
+
+}
+
+.editlink:hover {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: Use sass pseudo class syntax
```scss
.edit-link {
  &:hover { }
}
```

File: app/assets/stylesheets/group_members.scss [hunk 8]
```diff
@@ -1,3 +1,206 @@
...
+  border: 0;
+  border-radius: 5px;
+  color: $white;
+  cursor: pointer;
+  font-size: 16px;
+  left: 4px;
+  position: relative;
+  text-decoration: none;
+  top: -7px;
+
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: ✂️

File: app/assets/stylesheets/group_members.scss [hunk 9]
```diff
@@ -1,3 +1,206 @@
...
+}
+
+.addmem {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: Better naming please
Something like
```scss
div.add-member{
}
```

File: app/assets/stylesheets/group_members.scss [hunk 10]
```diff
@@ -1,3 +1,206 @@
...
+  cursor: pointer;
+  font-size: 16px;
+  padding-bottom: 12px;
+  padding-left: 34px;
+  padding-right: 34px;
+  padding-top: 12px;
+
+}
+
+.addmem:hover {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: Use sass pseudo class syntax as mentioned above

File: app/assets/stylesheets/group_members.scss [hunk 11]
```diff
@@ -1,3 +1,206 @@
...
+  border: 0;
+  border-radius: 15px;
+  color: $white;
+  cursor: pointer;
+  font-size: 16px;
+  text-decoration: none;
+
+}
+
+.view {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: View is generic name, can we have more specific name here ?

File: app/assets/stylesheets/group_members.scss [hunk 12]
```diff
@@ -1,3 +1,206 @@
...
+  cursor: pointer;
+  float: left;
+  font-size: 16px;
+  margin: 5px;
+  padding-bottom: 2px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 2px;
+  text-decoration: 0;
+
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: ✂️

File: app/assets/stylesheets/group_members.scss [hunk 13]
```diff
@@ -1,3 +1,206 @@
...
+}
+
+.view:hover {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: same as above

File: app/assets/stylesheets/group_members.scss [hunk 14]
```diff
@@ -1,3 +1,206 @@
...
+  color: $white;
+  text-decoration: none;
+
+}
+
+.editreopen {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: Better naming please

File: app/assets/stylesheets/group_members.scss [hunk 15]
```diff
@@ -1,3 +1,206 @@
...
+
+}
+
+.editreopen:hover {
+  color: $white;
+  text-decoration: none;
+
+}
+
+.del {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: Better naming please

File: app/views/groups/show.html.erb [hunk 16]
```diff
@@ -1,91 +1,65 @@
...
-                </tr>
-            <% end %>
-
-            <tbody/>
-          </table>
-
+      <br>
+      <br>
+      <h3>Members </h3>
+      <button type="button" class="float-sm-left float-md-right float-lg-right float-xl-right addmem text-white" data-toggle="modal" data-target="#myModal">Add member</button>
```
Comment (tachyons) [inline app/views/groups/show.html.erb]: #myModel is a placeholder name, have better naming here

**Comment by tachyons** [comment]
Good work so far, but not merge-able now because naming is poor, need better naming for css classes, otherwise maintenance will be hard  

```

    There are only two hard things in Computer Science: cache invalidation and naming things.

    -- Phil Karlton

```

File: app/assets/stylesheets/group_members.scss [hunk 17]
```diff
@@ -1,3 +1,211 @@
...
+  color: $white;
+  cursor: pointer;
+  font-size: 16px;
+  padding-bottom: 12px;
+  padding-left: 34px;
+  padding-right: 34px;
+  padding-top: 12px;
+}
+
+.add-member {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: merge this with above rule

File: app/assets/stylesheets/group_members.scss [hunk 18]
```diff
@@ -1,3 +1,211 @@
...
+  font-size: 16px;
+  margin: 5px;
+  padding-bottom: 2px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 2px;
+  text-decoration: 0;
+}
+
+.assignment-view {
```
Comment (tachyons) [inline app/assets/stylesheets/group_members.scss]: merge this with above rule

**Comment by darkshredder** [comment]
Does the class names look better @tachyons

**Pull request merged**